sources/CMLREAD

```diff
@@ -237,38 +240,41 @@ Copyright (c) 1985, 1986, 1987, 1988, 1990, 1993 by Venue & Xerox Corporation.
 
 (DECLARE%: EVAL@COMPILE
 
-(DATATYPE READER-ENVIRONMENT (REPACKAGE REREADTABLE REBASE RESPEC))
+(DATATYPE READER-ENVIRONMENT (REPACKAGE REREADTABLE REBASE NIL REFORMAT))
 )
 
-(/DECLAREDATATYPE 'READER-ENVIRONMENT '(POINTER POINTER POINTER POINTER)
+(/DECLAREDATATYPE 'READER-ENVIRONMENT '(POINTER POINTER POINTER POINTER POINTER)
        '((READER-ENVIRONMENT 0 POINTER)
          (READER-ENVIRONMENT 2 POINTER)
          (READER-ENVIRONMENT 4 POINTER)
-         (READER-ENVIRONMENT 6 POINTER))
-       '8)
+         (READER-ENVIRONMENT 6 POINTER)
+         (READER-ENVIRONMENT 8 POINTER))
+       '10)
 
 (DEFMACRO WITH-READER-ENVIRONMENT (ENV . BODY)
    `((CL:LAMBDA (E)
             (LET ((*PACKAGE* (ffetch (READER-ENVIRONMENT REPACKAGE) of E))
                   (*READTABLE* (ffetch (READER-ENVIRONMENT REREADTABLE) of E))
                   (*READ-BASE* (ffetch (READER-ENVIRONMENT REBASE) of E))
-                  (*PRINT-BASE* (ffetch (READER-ENVIRONMENT REBASE) of E)))
+                  (*PRINT-BASE* (ffetch (READER-ENVIRONMENT REBASE) of E))
+                  (*EXTERNALFORMAT* (ffetch (READER-ENVIRONMENT REFORMAT) of E)))
+                 (DECLARE (SPECVARS *EXTERNALFORMAT*))
                  ,@BODY))
      (\DTEST ,ENV 'READER-ENVIRONMENT)))
 
 (ADDTOVAR SYSSPECVARS *PACKAGE* *READTABLE* *READ-BASE* *PRINT-BASE*)
 
-(PUTPROPS WITH-READER-ENVIRONMENT INFO EVAL)
+(PUTPROPS WITH-READER-ENVIRONMENT INFO EVAL)
 (DECLARE%: DOEVAL@COMPILE DONTCOPY
 
 (GLOBALVARS *COMMON-LISP-READ-ENVIRONMENT*)
 )
 
 (RPAQ? *COMMON-LISP-READ-ENVIRONMENT* (create READER-ENVIRONMENT REPACKAGE _ (CL:FIND-PACKAGE
                                                                                   "USER")
-                                                 REREADTABLE _ CMLRDTBL REBASE _ 10))
```

Comment:
same ... common lisp vs. interlisp readtable shouldn't change the "stream" properties.